starter-code/scripts/controllers/articleController.js

```diff
@@ -43,6 +44,7 @@
   };
 
   // COMMENT: What does this method do?  What is it's execution path?
+  //This function is called on the root path.  It looks to see if the Article.all has items.  If yes then it saves the array as a ctx parameter and calls articlesController.index.  If no, then calls Article.fetchAll and passes it articleData as its callback. Article.data saves the articles as context parameters. And then calls articlesController.index. This is middleware for the root url.
```

Comment:
"... saves the articles as context parameters"
It's more accurate to say 'context property' or 'context variable'. Context parameters (or 'params') is a unique property on the context object that is generated from the parameters on the url path.